File: backend/src/main/java/com/example/user/application/port/in/UpdateUserUseCase.java
package com.example.user.application.port.in;

import com.example.user.application.command.UpdateUserCommand;
import com.example.user.application.command.UserResult;

/**
 * Updates an existing user.
 */
public interface UpdateUserUseCase {

    /**
     * Updates a user.
     *
     * @param command update command
     * @return updated user
     */
    UserResult execute(UpdateUserCommand command);
}

File: frontend/src/user/application/port/in/UpdateUserUseCase.ts
import { MutationUseCase } from '@/shared/application/port/in/MutationUseCase';
import { UserDto } from '@/user/domain/port/UserAdminGateway';
import { UpdateUserCommand } from '@/user/application/command/UpdateUserCommand';

export type UpdateUserUseCase = MutationUseCase<UpdateUserCommand, UserDto>;
